src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public ShopServiceImpl(Map<FruitTransaction.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void processTransactions(
+            List<FruitTransaction> transactions, Map<String, Integer> inventory) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationHandlers.get(transaction.getOperation());
+            if (handler != null) {
+                handler.handle(transaction, inventory);
+            } else {
+                throw new RuntimeException("Unknown operation: " + transaction.getOperation());
```

Comment:
When throwing a RuntimeException, it's better to include the cause of the exception as well. Since there's no underlying exception here, consider creating a specific exception class for unknown operations to provide more context for the error.